### PR TITLE
[LCAM-1745] Make passport assessment and financial assessments nullable

### PR DESCRIPTION
**Pull request by JHunt16**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/LCAM-1745)

Allow the passport assessment and financial assessments to be null in the determineMagsRepDecision request.

- Already made changes to the crime-commons library to update the ApiDetermineMagsRepDecisionRequest.
- Done some refactoring.
- Added a missing test scenario.

## Checklist

Before you ask people to review this PR:

- [ ] Tests should be passing: `./gradlew test`
- [ ] GitHub should not be reporting conflicts; you should have recently run `git rebase main`.
- [ ] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [ ] You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- [ ] You should have checked that the commit messages say why the change was made.

## Additional checks

- Don’t forget to [run](https://github.com/ministryofjustice/laa-crimeapps-maat-functional-tests/actions/workflows/ExecuteUiTests.yaml) the MAAT functional test suite after deploying your changes to the DEV or TEST environments to ensure your changes haven’t broken any of the functional tests.

### CODE REVIEW

File: crown-court-proceeding/src/test/java/uk/gov/justice/laa/crime/crowncourt/service/MagsProceedingServiceTest.java [hunk 1]
```diff
@@ -117,6 +117,15 @@ private static Stream<Arguments> getDecisionReasonScenarios() {
...
                 ),
+                Arguments.of(
+                      new Scenario(
+                          ReviewResult.PASS,
+                          null,
+                          null,
+                          null,
+                          PassportAssessmentResult.TEMP.getResult()
+                      ), DecisionReason.GRANTED
+                ),
```
Comment (Mhowell494) [inline crown-court-proceeding/src/test/java/uk/gov/justice/laa/crime/crowncourt/service/MagsProceedingServiceTest.java:128]: I think we're missing a few more test scenarios to cover the null checks you've added:
- No passport assessment & a financial assessment
- No financial assessment & no passport assessment
- A passport assessment & no financial assessment
- An `INEL` full assessment result

**Review by Mhowell494 — CHANGES_REQUESTED** [review]

File: crown-court-proceeding/src/test/java/uk/gov/justice/laa/crime/crowncourt/service/MagsProceedingServiceTest.java (hunk 1 above)
Comment (JHunt16) [reply crown-court-proceeding/src/test/java/uk/gov/justice/laa/crime/crowncourt/service/MagsProceedingServiceTest.java:145]: I've added some additional scenarios for 'No financial assessment & no passport assessment' and 'INEL'. We already have scenarios for the others you highlighted:

No passport assessment & a financial assessment - line 156, 174, 192 for example
A passport assessment & no financial assessment - line 120

Let me know if I've misunderstood though :)

Comment (Mhowell494) [reply crown-court-proceeding/src/test/java/uk/gov/justice/laa/crime/crowncourt/service/MagsProceedingServiceTest.java:145]: That's great - you're right we have test scenarios where we pass null for the financial and passported assessments, but this is just setting the result within the objects, but you've added in null checks for the objects themselves, which also needs to be tested.

**Review by Mhowell494 — APPROVED** [review]
